### PR TITLE
Python symbol/ndarray api re-org

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
preview: http://ec2-54-187-32-207.us-west-2.compute.amazonaws.com:999/api/python 

Summary:
- add one more level to NDArray API and Symbol API. Now there're separate pages for sparse NDArray/Symbol.
- add separate pages for random/linear algebra/contrib operators. 
- updated the docs for a few sparse operators

@reminisce @piiswrong

### CODE REVIEW

File: docs/api/python/index.md [hunk 1]
```diff
@@ -23,9 +23,14 @@ imported by running:
 
 ```eval_rst
 .. toctree::
+   :maxdepth: 3
```
Comment (piiswrong) [inline docs/api/python/index.md]: No need to list that many layers

**Comment by piiswrong** [comment]
break ndarray.md into two parts: NDArray Class and basic operators. Then add random, linalg etc

**Comment by eric-haibin-lin** [comment]
@rahul003 @bhavinthaker @reminisce @cjolivier01 @anirudh2290

File: docs/api/python/ndarray/ndarray.md [hunk 2]
```diff
@@ -625,16 +571,54 @@ We summarize the interface for each class in the following sections.
     Custom
 ```
 
+## Linear Algebra
```
Comment (piiswrong) [inline docs/api/python/ndarray/ndarray.md]: put linalg/rand/contrib in separate pages like sparse

Comment (eric-haibin-lin) [reply docs/api/python/ndarray/ndarray.md]: ok.. I misunderstood your previous comment.

**Comment by eric-haibin-lin** [comment]
Done. Also ping @reminisce

**Comment by asmushetzel** [comment]
Great that this is done now.

**Pull request merged**